### PR TITLE
Add `is.urlString`

### LINKED ISSUE #71: `is.urlString()`
Since checking if a given value is a valid URL is quite common, I think having a method such a method in `is` would be very useful (this could be done using [`is-url-superb`](https://github.com/sindresorhus/is-url-superb)).

I'm not sure if this directly aligns with the goals of the module, as a URL isn't really a data structure / type / object, but this method will still be useful in many cases. Also, one could argue that you could just use `is-url-superb` directly, but having this method in `is` would still be a good addition.

### PR DESCRIPTION
**Pull request by itaisteinherz**
This closes #71.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Looking at it now, I think we maybe should call it `is.urlString()` so it's not confused with checking whether it's an URL instance. When you see `is.url()` in some code it's not immediately clear whether it tests for `is.url(new URL())` or `is.url('http://example.com')`. Thoughts?

File: source/index.ts [hunk 1]
```diff
@@ -189,6 +189,19 @@ namespace is { // tslint:disable-line:no-namespace
...
 
+	export const url = (value: unknown) => {
+		if (!string(value)) {
+			return false;
+		}
+
+		try {
+			new URL(value); // tslint:disable-line no-unused-expression
+			return true;
+		} catch (_) { // tslint:disable-line no-unused
```
[the remark this answers is not in the document]
Comment (sindresorhus) [reply source/index.ts]:
```suggestion
		} catch {
```

You can just drop the catch-binding: https://blog.mariusschulz.com/2018/01/30/typescript-2-5-optional-catch-binding

[the remark this answers is not in the document]
Comment (itaisteinherz) [reply source/index.ts]: Cool!

**Comment by gioragutt** [comment]
@sindresorhus considering we have urlInstance, then urlString sounds like a good counterpart

**Comment by itaisteinherz** [comment]
> I think we maybe should call it `is.urlString()` so it's not confused with checking whether it's an URL instance. 

If we keep the current behavior then I agree we should rename the method to `is.urlString`. What I originally described in #71 is actually different than the method I have implemented here, since it returns `true` only if the given input is a `string` and is a valid URL.

I think it would be better to keep the method name the way it is and drop the `string` check.
Another option is to keep the behavior the way it is, rename the method to `is.urlString` and add another method `is.url`:
```typescript
const url = (value: unknown) => urlString(value) || urlInstance(value);
```

**Comment by sindresorhus** [comment]
> Another option is to keep the behavior the way it is, rename the method to is.urlString and add another method named is.url:

Hmm, I definitely think we should keep the behavior and rename to `urlString`, but not sure whether it makes sense to add a `is.url` method. Is it really a common use-case to need to check both at once? I would not imagine so. It would maybe be better to just leave it as two methods that you can combine if needed: `is.urlString(foo) && is.urlInstance(foo)`. In most cases you would just do `new URL(foo)` instead, which accepts either a string or `URL` instance.

**Comment by sindresorhus** [comment]
@itaisteinherz Can you rename it to `is.urlString` now and open a new issue about `is.url`? That discussion doesn't have to block this PR.

**Pull request merged**